### PR TITLE
feat: Integrate OpenAI audio models

### LINKED ISSUE #547: [Feature Request] integrate text to speech model
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [X] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

part of https://github.com/orgs/camel-ai/projects/1/views/2?pane=issue&itemId=32191111

### Solution

https://platform.openai.com/docs/guides/text-to-speech

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Wendong-Fan**
## Description

Integrate OpenAI's text to speech and speech to text models

1. text_to_speech using `tts-1`, with chunking module to solve the character limitation 4096
2. speech_to_text using `whisper-1`, with chunking module to solve the size limitation 25mb

## Motivation and Context

`close #547 `

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/models/openai_audio_models.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+        self._client = OpenAI(timeout=120, max_retries=3, base_url=url)
+
+    def text_to_speech(
+        self,
+        input: str,
+        model_type: AudioModelType = AudioModelType.TTS_1,
+        voice: VoiceType = VoiceType.ALLOY,
+        **kwargs: Any,
+    ) -> Union[
+        List[_legacy_response.HttpxBinaryResponseContent],
```
Comment (zechengz) [inline camel/models/openai_audio_models.py:40]: It seems that this "legacy" response will be deprecated soon. Do we have any other return type?

File: camel/models/openai_audio_models.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            # Create new chunk path
+            chunk_path = os.path.join(output_dir, f"chunk_{i}.{audio_format}")
+            chunk.export(chunk_path, format=audio_format)
+            split_files.append(chunk_path)
+        return split_files
+
+    def speech_to_text(
+        self,
+        audio_file_path: str,
+        translate_into_eng: bool = False,
```
Comment (zechengz) [inline camel/models/openai_audio_models.py:152]: nit: `translate_into_english`

File: camel/models/openai_audio_models.py (hunk 1 above)
Comment (Wendong-Fan) [reply camel/models/openai_audio_models.py:40]: I didn't find alternative type from OpenAI, when legacy is deprecated I think they should introduce a new type, let's do the update after openai's update~

File: camel/models/openai_audio_models.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+                single response content if input charaters less than 4096.
+
+        Raises:
+            Exception: If there's an error during the TTS API call.
+        """
+        try:
+            # Model only support at most 4096 characters one time.
+            max_chunk_size = 4095
+            audio_chunks = []
+            if len(input) > max_chunk_size:
```
[the remark this answers is not in the document]
Comment (Wendong-Fan) [reply camel/models/openai_audio_models.py:68]: Hey @raywhoelse  thanks for the review! The limitation is not token but characters, refer to https://help.openai.com/en/articles/8555505-tts-api

File: camel/models/openai_audio_models.py [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+                        chunk = input[:max_chunk_size]
+                        input = input[max_chunk_size:].lstrip()
+
+                    response = self._client.audio.speech.create(
+                        model=model_type.value,
+                        voice=voice.value,
+                        input=chunk,
+                        **kwargs,
+                    )
+                    audio_chunks.append(response)
```
[the remark this answers is not in the document]
Comment (Wendong-Fan) [reply camel/models/openai_audio_models.py:91]: Great point, added feature allow user to input storage path to store the generated file

**Review by zechengz — APPROVED** [review]

**Pull request merged**